src/java/org/apache/cassandra/repair/state/ParticipateState.java

```diff
@@ -59,10 +64,30 @@ public ParticipateState(InetAddressAndPort initiator, PrepareMessage msg)
 
     public boolean register(ValidationState state)
     {
+        maybeUpdateState(State.VALIDATION);
         ValidationState current = validations.putIfAbsent(state.id, state);
         return current == null;
     }
 
+    @Nullable
+    public ValidationState validation(UUID id)
+    {
+        return validations.get(id);
+    }
+
+    public boolean register(SyncState state)
+    {
+        maybeUpdateState(State.SYNC);
```

Comment:
this looks wrong - VALIDATION and SYNC is per-table in the parent repair session, so with this we flip back and forth, which makes basically all the new tests fail

Reply:
yeah, I rushed a refactor and that broke tests, I just pushed a fix that makes this nested

```
jobs:
  desc:
    validation: state
    syncs: map<id, state>
```